tests/Admin/Filter/CategoryFilterTest.php

```diff
@@ -41,6 +44,37 @@ public function testRenderSettings(): void
         $options = $filter->getRenderSettings()[1];
 
         $this->assertSame(ChoiceType::class, $options['field_type']);
-        $this->assertSame([], $options['field_options']['choices']);
+        $this->assertSame([
+            $category,
+        ], $options['field_options']['choices']);
+    }
+
+    public function testRenderSettingsWithContext(): void
+    {
+        $this->categoryManager->method('getCategories')->willReturn([
+            $category = $this->createCategoryMock(),
+        ]);
+
+        $filter = new CategoryFilter($this->categoryManager);
+        $filter->initialize('field_name', [
+            'field_options' => [
+                'class' => 'FooBar',
+                'context' => 'foo',
+            ],
+        ]);
+        $options = $filter->getRenderSettings()[1];
+
+        $this->assertSame(ChoiceType::class, $options['field_type']);
+        $this->assertSame([
+            $category,
+        ], $options['field_options']['choices']);
+    }
+
+    private function createCategoryMock(): CategoryInterface
+    {
+        $category = $this->createStub(CategoryInterface::class);
```

Comment:
if you are mocking methods, you should use create mock instead right?

Reply:
Not exactly right, basically if there is no calls to `expects()` method, it is a stub (https://github.com/sebastianbergmann/phpunit-mock-objects/issues/411).